---
title: "Genome Assembly Assessment"
output: html_document
bibliography: "../manuscript/references.bib"
---

# Exploratory Data Analysis

As part of researching alternatives to N50, exploratory data analsysis was 
conducted using factors about genome assembly quality from 797 vertebrate
genomes from [NCBI](https://www.ncbi.nlm.nih.gov/genome). In this document,
I explore the relationships between these factors, particularly focusing on
the relationship between factors describing contiguity and completeness and
contiguity and correctness.

## Setting up

The following libraries and functions are necessary for this analsyis.
```{r message = FALSE, set-up}
library(knitr)
library(dplyr)
library(Hmisc)
library(reshape2)
library(ggplot2)
library(scales)
library(xtable)

# Source: 
# https://rstudio-pubs-static.s3.amazonaws.com/240657_5157ff98e8204c358b2118fa69162e18.html
flatten_correlation_matrix <- function(cor_r, cor_p){
  #This function provides a simple formatting of a correlation matrix
  #into a table with 4 columns containing :
    # Column 1 : row names (variable 1 for the correlation test)
    # Column 2 : column names (variable 2 for the correlation test)
    # Column 3 : the correlation coefficients
    # Column 4 : the p-values of the correlations
  library(tidyr)
  library(tibble)
  cor_r <- rownames_to_column(as.data.frame(cor_r), var = "row")
  cor_r <- gather(cor_r, column, cor, -1)
  cor_p <- rownames_to_column(as.data.frame(cor_p), var = "row")
  cor_p <- gather(cor_p, column, p, -1)
  cor_p_matrix <- left_join(cor_r, cor_p, by = c("row", "column"))
  cor_p_matrix
}
```

## Data Description

These statistics were obtained by downloading all vertebrate genomes from NCBI
and analyzing the assemblies with 
[abyss-fac](https://github.com/bcgsc/abyss/blob/master/bin/abyss-fac.pl) and 
[BUSCO](https://busco.ezlab.org/) on a computing cluster. A BASH 
script was used to format the output of these two programs as a single, 
tab-delimited file. The data represented by the columns is described as follows:

name
: the unique name of the assembly

N50
: the length of the smallest contig such that 50% of the genome is contained in
contigs of size N50 or larger, describes genome contiguity

e-size
: the length of the contig containing a randomly selected base in the genome,
describes genome contiguity

num_contigs
: the number of contigs in an assembly, describes genome contiguity

complete
: the number of complete single-copy BUSCOs, describes genome completeness

fragmented
: the number of fragmented BUSCOs, describes genome completeness

missing
: the number of missing BUSCOs, describes genome completeness

duplicate
: the number of duplicated BUSCOs, describes genome correctness

```{r data}
data = read.table("../data/features.tsv")
colnames(data) = c("name", "N50", "esize", "num_contigs", "complete",
                   "fragmented", "missing", "duplicate")
```

## Correlation

To explore the correlation of the various factors, a correlation matrix using
Spearman's correlation. Non-significant coefficients are replaced with `NA`. 
Correlations between contiguity (N50, e-size, and num_contigs) and the other 
factors from BUSCO are particularly interesting; correlations that represent contiguity-contiguity and BUSCO-BUSCO will be removed in the plot.

```{r correlation_matrix}
kable(rcorr(as.matrix(data %>% select(-name)), type="spearman")$r)
```

```{r plot}
correlation_matrix = rcorr(as.matrix(data %>% select(-name)), type="spearman")
correlation_matrix = flatten_correlation_matrix(correlation_matrix$r,
                                                correlation_matrix$P)
correlation_matrix =  correlation_matrix %>% 
    mutate(cor = ifelse(p >= 0.05,
           NA,
           .data$cor))
ggplot(data = correlation_matrix %>% filter(row != "num_contigs", 
                                            row != "esize", 
                                            row != "N50",
                                            column != "complete",
                                            column != "fragmented",
                                            column != "missing",
                                            column != "duplicate"), 
       aes(column, row, fill = cor))+
    geom_tile(color = "white") +
    theme_minimal() + 
    scale_fill_gradient2(low = muted("red"),
                         mid = "white", 
                         high = muted("blue"), 
                         midpoint = 0, 
                         space = "Lab",
                         na.value = "grey50",
                         guide = "colourbar",
                         aesthetics = "fill",
                         name="Spearman\nCorrelation") +
    theme(axis.text.x = element_text(angle = 45, 
                                     vjust = 1,
                                     size = 12,
                                     hjust = 1)) +
    geom_text(aes(label=round(cor, 3)) , color="black") + 
    labs(x = "Contiguity", y = "Completeness & Correctness")
ggsave("../manuscript/imgs/1.png", width = 11, height = 8.5, dpi = 600)
```

Of these comparisons, a few stand out. Notably, none of these factors are
strongly correlated, which is to be expected [@Salzberg2012; @Utturkar2014]. 
The strongest factors are all related to the number of fragmented BUSCOs. This 
relationship can be explained by considering the the relationship between 
contiguity and fragmented BUSCOs --- as the number of contigs decreases, any
BUSCOs that might be fragmented across contigs are instead now joined. Some
relationship between contiguity and fragmented genes is expected 
[@grau_improving_2018].

```{r fragmented_vs_N50}
ggplot(data) + geom_point(aes(N50, fragmented), alpha = 0.5)
```

The two outliers make this plot somewhat hard to see. Investigating them, we
find that one of them seems to have no interesting features. To retain the 
second outlier, a filter of 5e8 was added for e-size, and the plot was
recreated. These outliers are ignored going forward.

```{r outliers}
kable(data %>% filter(N50 > 2e8) %>% select(-name))

ggplot(data %>% filter(N50 < 2e8)) + 
    geom_point(aes(N50, fragmented), alpha = 0.5)
```

The next strongest correlations are those of the percent of complete BUSCOs with
N50, e-size, and the number of contigs, and their correlations with the percent
of missing BUSCOs is close behind. A scatterplot comparing these features is
informative. 

```{r complete_vs_N50}
ggplot(data %>% filter(N50 < 2e8), aes(N50, complete)) + 
    geom_point(alpha = 0.5)
ggsave("../manuscript/imgs/3.png", width = 11, height = 8.5, dpi = 600)
```
  
Deciphering a relationship between N50 and the percentage of complete BUSCOs is
difficult based on the information provided in this graph. There are many
genomes with high completeness with N50s ranging along the entire x-axis. In 
examining the y-axis, there are many genomes with low N50 and percentage of 
complete BUSCOs ranging along the entire y-axis.

In other words, given a low N50, no prediction about the percentage of complete
BUSCOs can be made. Given a high N50, a high percentage of complete BUSCOs
seems more likely but is not guaranteed. The inverse is also true --- given a
low percent complete, the N50 is likely to also be low. Given a high
percent complete, N50 can range from 0 to 1.5e8.

## Towards a New Metric

Here we check correlations among contiguity features and BUSCO features to see which, if any, can be dropped due to high correlation with another feature.

```{r other_correlations}
ggplot(data = correlation_matrix %>% filter(row != "complete", 
                                            row != "fragmented", 
                                            row != "missing",
                                            row != "duplicate",
                                            column != "complete",
                                            column != "fragmented",
                                            column != "missing",
                                            column != "duplicate"), 
       aes(column, row, fill = cor))+
    geom_tile(color = "white") +
    theme_minimal() + 
    scale_fill_gradient2(low = muted("red"),
                         mid = "white", 
                         high = muted("blue"), 
                         midpoint = 0, 
                         space = "Lab",
                         na.value = "grey50",
                         guide = "colourbar",
                         aesthetics = "fill",
                         name="Spearman\nCorrelation") +
    theme(axis.text.x = element_text(angle = 45, 
                                     vjust = 1,
                                     size = 12,
                                     hjust = 1)) +
    geom_text(aes(label=round(cor, 3)) , color="black") + 
    labs(x = "Contiguity features", y = "Contiguity features")
ggsave("../manuscript/imgs/contiguity_correlation.png", width = 11, height = 8.5, dpi = 600)

ggplot(data = correlation_matrix %>% filter(row != "num_contigs", 
                                            row != "esize", 
                                            row != "N50",
                                            column != "num_contigs",
                                            column != "esize",
                                            column != "N50"), 
       aes(column, row, fill = cor))+
    geom_tile(color = "white") +
    theme_minimal() + 
    scale_fill_gradient2(low = muted("red"),
                         mid = "white", 
                         high = muted("blue"), 
                         midpoint = 0, 
                         space = "Lab",
                         na.value = "grey50",
                         guide = "colourbar",
                         aesthetics = "fill",
                         name="Spearman\nCorrelation") +
    theme(axis.text.x = element_text(angle = 45, 
                                     vjust = 1,
                                     size = 12,
                                     hjust = 1)) +
    geom_text(aes(label=round(cor, 3)) , color="black") + 
    labs(x = "BUSCO features", y = "BUSCO features")
ggsave("../manuscript/imgs/2.png", width = 11, height = 8.5, dpi = 600)
```

The distance to the ideal was created as a way to encode expert knowledge in a metric. The table with number of chromosomes is available for download at https://genomesize.com, but this data has to be downloaded by each individual; the terms of use don't allow redistribution. The features were also renamed to match the formatting of the animal size database.

```{r distance}
distance_to_ideal = function(num_contigs, num_chr, complete, duplicate) {
    sqrt((100-complete)^2 + (100-(num_chr/num_contigs)*100)^2 + (0 - duplicate)^2)
}


data = read.table("../data/genome_size_data_130519_09_32_45.csv", header=TRUE, sep=",",
                  quote = "\"") %>%
    select(Species, C.value, Chrom.num) %>%
    mutate(Species = as.character.factor(Species),
           Chrom.num = as.numeric(levels(Chrom.num))[Chrom.num],
           C.value = as.numeric(levels(C.value))[C.value],
           size = (0.978 * 10e9)*C.value) %>% 
    filter(!is.na(Chrom.num))

rankings = read.table("../data/renamed.features.csv", sep=",")
colnames(rankings) = c("Species", "N50", "esize", "num_contigs", "complete",
                       "fragmented", "missing", "duplicate")
merged = merge(data, rankings, by = "Species") %>%
    rowwise() %>%
    mutate(distance = distance_to_ideal(num_contigs, 
                                        Chrom.num/2, 
                                        complete,
                                        duplicate))

sampled = rbind(head(merged %>% arrange(distance), 5), 
                merged[sample(nrow(merged), 20), ],
                tail(merged %>% arrange(distance), 5)) %>% 
    select(Species, N50, num_contigs, complete, duplicate, distance)
xtable(sampled %>% arrange(distance))
```